File: sdk/js-vault-sdk/src/services/audit.ts
import type { VaultClient } from "../client";
import type { AuditList, AuditListOptions } from "../types/audit";

export class AuditService {
  constructor(private readonly client: VaultClient) {}

  public async list(options: AuditListOptions = {}): Promise<AuditList> {
    const params: string[] = [];
    if (options.secretId) params.push(`secret_id=${encodeURIComponent(options.secretId)}`);
    if (options.action) params.push(`action=${encodeURIComponent(options.action)}`);
    if (options.actorId) params.push(`actor_id=${encodeURIComponent(options.actorId)}`);
    if (options.limit !== undefined) params.push(`limit=${options.limit}`);
    if (options.offset !== undefined) params.push(`offset=${options.offset}`);
    const qs = params.length ? `?${params.join("&")}` : "";
    return this.client.request<AuditList>("GET", `/v1/vault/audit${qs}`);
  }
}

File: sdk/python-vault-sdk/functionfly_vault/services/audit.py
from __future__ import annotations

from typing import Any, Dict, List, Optional
from urllib.parse import urlencode


class AuditService:
    def __init__(self, client: Any) -> None:
        self._client = client

    def list(
        self,
        secret_id: Optional[str] = None,
        action: Optional[str] = None,
        actor_id: Optional[str] = None,
        limit: int = 50,
        offset: int = 0,
    ) -> Dict[str, Any]:
        params: Dict[str, Any] = {"limit": limit, "offset": offset}
        if secret_id is not None:
            params["secret_id"] = secret_id
        if action is not None:
            params["action"] = action
        if actor_id is not None:
            params["actor_id"] = actor_id
        return self._client.request("GET", "/v1/vault/audit?" + urlencode(params))
